server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/filter/MailMatcher.java

```diff
@@ -89,23 +89,27 @@ public boolean match(Mail mail) {
                 .build();
 
         static ContentMatcher negate(ContentMatcher contentMatcher) {
-            return (String fullContent, String matchingValue) ->
-                    !contentMatcher.match(fullContent, matchingValue);
+            return (Stream<String> contents, String valueToMatch) ->
```

Comment:
by using Stream<String> we can avoid the inconsistent between `CONTAINS` & `NOT CONTAINS` or `EXACTLY_EQUALS` & `NOT_EXACTLY_EQUALS`.  

Before that we single string matching which lead to in some case with same mail, `CONTAINS` & `NOT CONTAINS` rule all match that mail while expecting only one got matched